src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -1356,158 +1308,19 @@ public static List<(char First, char Last)>? ComputeRanges(ReadOnlySpan<char> se
-                    case '\u0018': return "\0\u0002\0\u0018\u0019";
-                    case '\u0019': return "\0\u0002\0\u0019\u001A";
-                    case '\u001A': return "\0\u0002\0\u001A\u001B";
-                    case '\u001B': return "\0\u0002\0\u001B\u001C";
-                    case '\u001C': return "\0\u0002\0\u001C\u001D";
-                    case '\u001D': return "\0\u0002\0\u001D\u001E";
-                    case '\u001E': return "\0\u0002\0\u001E\u001F";
-                    case '\u001F': return "\0\u0002\0\u001F\u0020";
-                    case '\u0020': return "\0\u0002\0\u0020\u0021";
-                    case '\u0021': return "\0\u0002\0\u0021\u0022";
-                    case '\u0022': return "\0\u0002\0\u0022\u0023";
-                    case '\u0023': return "\0\u0002\0\u0023\u0024";
-                    case '\u0025': return "\0\u0002\0\u0025\u0026";
-                    case '\u0026': return "\0\u0002\0\u0026\u0027";
-                    case '\u0027': return "\0\u0002\0\u0027\u0028";
-                    case '\u0028': return "\0\u0002\0\u0028\u0029";
-                    case '\u0029': return "\0\u0002\0\u0029\u002A";
-                    case '\u002A': return "\0\u0002\0\u002A\u002B";
-                    case '\u002C': return "\0\u0002\0\u002C\u002D";
-                    case '\u002D': return "\0\u0002\0\u002D\u002E";
-                    case '\u002E': return "\0\u0002\0\u002E\u002F";
-                    case '\u002F': return "\0\u0002\0\u002F\u0030";
-                    case '\u0030': return "\0\u0002\0\u0030\u0031";
-                    case '\u0031': return "\0\u0002\0\u0031\u0032";
-                    case '\u0032': return "\0\u0002\0\u0032\u0033";
-                    case '\u0033': return "\0\u0002\0\u0033\u0034";
-                    case '\u0034': return "\0\u0002\0\u0034\u0035";
-                    case '\u0035': return "\0\u0002\0\u0035\u0036";
-                    case '\u0036': return "\0\u0002\0\u0036\u0037";
-                    case '\u0037': return "\0\u0002\0\u0037\u0038";
-                    case '\u0038': return "\0\u0002\0\u0038\u0039";
-                    case '\u0039': return "\0\u0002\0\u0039\u003A";
-                    case '\u003A': return "\0\u0002\0\u003A\u003B";
-                    case '\u003B': return "\0\u0002\0\u003B\u003C";
-                    case '\u003F': return "\0\u0002\0\u003F\u0040";
-                    case '\u0040': return "\0\u0002\0\u0040\u0041";
-                    case '\u005B': return "\0\u0002\0\u005B\u005C";
-                    case '\u005C': return "\0\u0002\0\u005C\u005D";
-                    case '\u005D': return "\0\u0002\0\u005D\u005E";
-                    case '\u005F': return "\0\u0002\0\u005F\u0060";
-                    case '\u007B': return "\0\u0002\0\u007B\u007C";
-                    case '\u007D': return "\0\u0002\0\u007D\u007E";
-                    case '\u007F': return "\0\u0002\0\u007F\u0080";
```

Comment:
This isn't necessary any longer... but I wonder if an allocation trace of `new Regex("some pattern with lots of letters", RegexOptions.IgnoreCase)` would show any benefits to keeping it?

Reply:
I will run some perf tests and see how is construction time being impacted by my current implementation, and see if we may want to special case ASCII letters if it is.

Reply:
so I ran the following benchmarks before and after my changes:

```c#
[Benchmark] public void ManyCharactersTestCtor() => new Regex("some pattern with lots of letters and words and more and more", Options);
[Benchmark] public void ManyCharactersTestNoMatch() => new Regex("some pattern with lots of letters and words and more and more", Options).IsMatch(_loremIpsum);
[Benchmark] public void ManyCharactersTestMatch() => new Regex("some pattern with lots of letters and words and more and more", Options).IsMatch("SOME PATTERN WITH LOTS OF LETTERS AND WORDS AND MORE AND MORE");
```

Here are the results for Options = `RegexOptions.Compiled | RegexOptions.IgnoreCase`:

|                    Method |        Job |         Toolchain |              Options |            Mean |         Error |        StdDev |          Median |             Min |             Max | Ratio | RatioSD |  Gen 0 |  Gen 1 | Allocated | Alloc Ratio |
|-------------------------- |----------- |------------------ |--------------------- |----------------:|--------------:|--------------:|----------------:|----------------:|----------------:|------:|--------:|-------:|-------:|----------:|------------:|
|    ManyCharactersTestCtor | Job-NCBHSR | \main\corerun.exe | IgnoreCase, Compiled |    23,152.15 ns |    535.172 ns |    616.305 ns |    23,033.33 ns |    22,211.52 ns |    24,618.60 ns |  1.00 |    0.00 | 3.0401 | 0.0894 |   19208 B |        1.00 |
|    ManyCharactersTestCtor | Job-WYSVXN |   \pr\corerun.exe | IgnoreCase, Compiled |    22,600.47 ns |    270.913 ns |    211.511 ns |    22,615.91 ns |    22,283.50 ns |    22,905.56 ns |  0.98 |    0.03 | 3.2562 | 0.0904 |   20656 B |        1.08 |
|                           |            |                   |                      |                 |               |               |                 |                 |                 |       |         |        |        |           |             |
| ManyCharactersTestNoMatch | Job-NCBHSR | \main\corerun.exe | IgnoreCase, Compiled |   772,878.90 ns | 15,302.233 ns | 17,008.392 ns |   776,661.56 ns |   748,841.25 ns |   803,732.19 ns |  1.00 |    0.00 | 3.1250 |      - |   21188 B |        1.00 |
| ManyCharactersTestNoMatch | Job-WYSVXN |   \pr\corerun.exe | IgnoreCase, Compiled |   759,382.98 ns |  8,814.636 ns |  8,245.216 ns |   758,668.44 ns |   743,135.62 ns |   775,794.06 ns |  0.98 |    0.02 | 3.1250 |      - |   22624 B |        1.07 |
|                           |            |                   |                      |                 |               |               |                 |                 |                 |       |         |        |        |           |             |
|   ManyCharactersTestMatch | Job-NCBHSR | \main\corerun.exe | IgnoreCase, Compiled | 1,883,317.63 ns | 13,621.784 ns | 12,075.356 ns | 1,883,850.39 ns | 1,863,339.06 ns | 1,904,969.53 ns |  1.00 |    0.00 |      - |      - |   21124 B |        1.00 |
|   ManyCharactersTestMatch | Job-WYSVXN |   \pr\corerun.exe | IgnoreCase, Compiled | 1,841,654.17 ns | 18,595.733 ns | 16,484.633 ns | 1,836,657.99 ns | 1,814,689.58 ns | 1,866,917.36 ns |  0.98 |    0.01 |      - |      - |   22564 B |        1.07 |

So I'm not seeing a regression, just a very slight improvement for that case.

Reply:
That allocation column highlights a regression.

As a quick hack, I pulled down your change, built it, then added this into the CharsToStringClass method:
```C#
            if (chars.Length == 2)
            {
                switch (chars[0], chars[1])
                {
                    case ('A', 'a'): case ('a', 'A'):return "\0\x0004\0ABab";
                    case ('B', 'b'): case ('b', 'B'):return "\0\x0004\0BCbc";
                    case ('C', 'c'): case ('c', 'C'):return "\0\x0004\0CDcd";
                    case ('D', 'd'): case ('d', 'D'):return "\0\x0004\0DEde";
                    case ('E', 'e'): case ('e', 'E'):return "\0\x0004\0EFef";
                    case ('F', 'f'): case ('f', 'F'):return "\0\x0004\0FGfg";
                    case ('G', 'g'): case ('g', 'G'):return "\0\x0004\0GHgh";
                    case ('H', 'h'): case ('h', 'H'):return "\0\x0004\0HIhi";
                    case ('J', 'j'): case ('j', 'J'):return "\0\x0004\0JKjk";
                    case ('L', 'l'): case ('l', 'L'):return "\0\x0004\0LMlm";
                    case ('M', 'm'): case ('m', 'M'):return "\0\x0004\0MNmn";
                    case ('N', 'n'): case ('n', 'N'):return "\0\x0004\0NOno";
                    case ('O', 'o'): case ('o', 'O'):return "\0\x0004\0OPop";
                    case ('P', 'p'): case ('p', 'P'):return "\0\x0004\0PQpq";
                    case ('Q', 'q'): case ('q', 'Q'):return "\0\x0004\0QRqr";
                    case ('R', 'r'): case ('r', 'R'):return "\0\x0004\0RSrs";
                    case ('S', 's'): case ('s', 'S'):return "\0\x0004\0STst";
                    case ('T', 't'): case ('t', 'T'):return "\0\x0004\0TUtu";
                    case ('U', 'u'): case ('u', 'U'):return "\0\x0004\0UVuv";
                    case ('V', 'v'): case ('v', 'V'):return "\0\x0004\0VWvw";
                    case ('W', 'w'): case ('w', 'W'):return "\0\x0004\0WXwx";
                    case ('X', 'x'): case ('x', 'X'):return "\0\x0004\0XYxy";
                    case ('Y', 'y'): case ('y', 'Y'):return "\0\x0004\0YZyz";
                    case ('Z', 'z'): case ('z', 'Z'): return "\0\x0004\0Z[z{";
                }
            }
```
built it again, and ran your ManyCharactersTestCtor with RegexOptions.IgnoreCase (before here is your change, after is your change plus the above if block):

|                 Method |         Toolchain |    Options |     Mean |    Error |   StdDev | Ratio |  Gen 0 | Allocated | Alloc Ratio |
|----------------------- |------------------ |----------- |---------:|---------:|---------:|------:|-------:|----------:|------------:|
| ManyCharactersTestCtor | \before\corerun.exe | IgnoreCase | 15.20 us | 0.247 us | 0.231 us |  1.00 | 0.1831 |  14.32 KB |        1.00 |
| ManyCharactersTestCtor |   \after\corerun.exe | IgnoreCase | 14.48 us | 0.109 us | 0.102 us |  0.95 | 0.1526 |  12.41 KB |        0.87 |